### DISCUSSION TITLE
JS code not working inside swup container

### DISCUSSION THREAD

**Discussion by sourabh0404** [swup/swup#700] [Help]
<main id="swup" class="transition-fade"> </main>
inside this code there are multiple sections with js animation used gsap for that. So the issue is when i get on next page smoothly with swup then all the js animation does not work on that page until i refresh the page. How i can fix this issue?

**Comment by daun** [comment]
Hi! Have you found a solution to your problem? The docs on [Reloading JavaScript](https://swup.js.org/getting-started/reloading-javascript/) should get you pretty far.

**Reply by sourabh0404** [reply]
Yes Philip that helped me alot thanx! 
But i have one more question so the thing is i created 4 pages total in vs code and when i open each page separately its working well. After adding swup when i open home page and get into new page from menu then it is also working well without any issue new. But the issue is except home page when i directly open some other page like contact and if i try to get into another page from contact's menu, then that particular is not working well by that i mean is no any css style even i added that css file link , but as i refresh the page then css adds to the page automatically, how i can fix this issue?

**Reply by daun** [reply]
We recommend that swup users either a) put all their styles into a single stylesheet or b) install the [Head Plugin](https://swup.js.org/plugins/head-plugin/) and enable its [awaitAssets](https://swup.js.org/plugins/head-plugin/#await-assets) option so it will wait for new styles to load.

**Reply by sourabh0404** [reply]
You are helping me alot! thanx alot for that man, promise this is gonna be my last question
Can you please  look at this code im having alot problem 
Both page-container and home-page are different pages

```js
const swup = new Swup();
function init() {
  if(document.querySelector('#page-container')){


    gsap.registerPlugin(ScrollTrigger);

    let sections = gsap.utils.toArray(".panel");
    
    gsap.to(sections, {
      xPercent: -85 * (sections.length - 1),
      ease: "none",
      scrollTrigger: {
        trigger: ".container",
        pin: true,
        scrub: 1,
      }
    });
  
  }
  if (document.querySelector('#home-page')) {

    const lenis = new Lenis({
      duration: 1.2,
      easing: (t) => Math.min(1, 1.001 - Math.pow(2, -10 * t))
    })
    
    function raf(time) {
        lenis.raf(time)
        requestAnimationFrame(raf)
    }
    
    requestAnimationFrame(raf)

}
}

if (document.readyState === 'complete') {
  init();
} else {
  document.addEventListener('DOMContentLoaded', () => init());
}

swup.on('pageView', () => init());
```

**Reply by sourabh0404** [reply]
issue is home-page code is also working in page-con page but when i open directly the page-con(in vs code go live) then home-page code is not working in this page but when i get into hom-page from page-con then home-page code is not working overall its all messy

**Reply by hirasso** [reply]
@sourabh0404 you should format your code snippets using three backticks, followed by the language, like so:
~~~
```js
const foo = "bar";
```
~~~
...will be rendered like this:
```js
const foo = "bar";
```

**Reply by daun** [reply]
@sourabh0404 I'm not sure I understand what issue you're seeing. Without knowing how and where you're loading the script, this is impossible to debug properly.

**Reply by daun** [reply]
@sourabh0404 I don't think this is related to your issue, but just as an aside: if you're starting animation loops in `pageView`, you must make sure to correctly stop or remove them before a new page is loaded, i.e. in `willReplaceContent`. See the docs on [Avoiding memory leaks](https://swup.js.org/getting-started/reloading-javascript/#avoiding-memory-leaks) for an example.

**Reply by sourabh0404** [reply]
https://github.com/sourabh0404/web.git

sorry for bethering if you have time you can check this out (main.js in js file) im trying to reload javascript code, there are multiple pages so its kinda confusing to me
